### ISSUE TITLE
[Request] Alternative SocialSpy output - raw broadcast.

### ISSUE THREAD

**Issue by mibby** [EssentialsX/Essentials#788]
Is it possible to add a setting to go back to the old socialspy formatting and display? A lot of my staff do not like the new socialspy output, even with the prefix removed. Everyone far prefers the raw broadcast.

**Comment by mibby** [comment]
@SupaHam Or alternatively, perhaps a way so we can customize the social spy formatting? For example changing bracket colors, having raw username (not colorized / displayed with prefix), and message format after the user -> user display? 

Such as if you wanted the entire socialspy display light gray so it doesn't stand out or conflict too much with normal chat color.

The current output is very messy - lots of complaints from staff that it looks like their own private messages so they are forced to read everything / miss their own messages rather than it being a background tool for monitoring. I think it would be better if there was a setting to go back to the raw broadcast display or if there was the ability to customize socialspy output separately and have it entirely dark/light gray.

**Comment by SupaHam** [comment]
The prefix can be removed: https://github.com/drtshock/Essentials/blob/2.x/Essentials/src/messages.properties#L412

**Comment by mibby** [comment]
I'm aware the prefix can be removed, as noted above, but it does not help much. Many of my staff are voicing their complaints with the new social spy and getting confused with all the message output - unsure if they are receiving messages due to everything blending in.

I'm requesting either a toggle to go back to the raw broadcast or a way to format social spy output separately so I could make the entire display dark gray for example so it isn't blending in with normal chat / serves more as a background utility for moderating. :)

**Comment by mibby** [comment]
Bump. ^^

Raw output toggle or possibility to format socialspy display output separately from normal message output?

**Comment by Kealper** [comment]
I'd like a way to change the formatting of the messages themselves or go back to the raw commands as well. The [SS] tag isn't the problem, the problem is that the current formatting blends in with the regular chat too much overall. If customizable formatting were an option, I'd personally probably just make the whole line gray, message body included.

**Comment by Momshroom** [comment]
You can edit it to make the entire line grey if you open up the jar.   I'd
like the raw commands back b/c some things that DID show in the raw
commands no longer show.

**Comment by Anesitas** [comment]
When I try to make the line grey, all msg chat goes raw. what line did you use then?

**Comment by mibby** [comment]
@SupaHam Any update to this issue? Socialspy is completely unusable when player names are prefixed colored. It is a wall of color with everyone messaging each other.

I'd like to have an option for raw message output in socialspy like before or a completely separate language string specifically for socialspy formatting to be able to make the entire thing grey without having to change the normal msgFormat which affects standard people messaging others.

**Comment by SupaHam** [comment]
Will look into this soon™. :)

edit: ™

**Comment by SupaHam** [comment]
What I have currently:
socialSpyMsgFormat=\u00a7f[\u00a76SS\u00a7f] \u00a76[\u00a7c{0}\u00a76 -> \u00a7c{1}\u00a76] \u00a7r {2}

{0} is sender player name, {1} recipient player name, {2} message.

`socialSpyMsgFormat` would have its own special case just for the messaging commands. So it can be tailored to your specifications. Is this a sufficient solution?

**Comment by mdcfe** [comment]
@SupaHam #1226 already seems to implement this.

**Comment by SupaHam** [comment]
@mibby I'm still waiting to hear back, before resolving the issue.

**Comment by mibby** [comment]
> Is this a sufficient solution?

@SupaHam Would the sender and recipient player names be sent raw or based on their vault prefix / displayname? As I'd personally like to colorize the entire message string dark grey. It would look a bit silly if the entire thing was dark grey but the names themselves. 

Prior socialspy behavior sent the entire message in a raw broadcast so it wasn't as obtrusive with blending in with normal chat.

Edit: Though I guess it is something I could live with if raw output isn't something that can be supported, since I could just change the bracket and the message itself to grey and live with the colored names within the bracket. It would be something better than it behaves now.

**Comment by mibby** [comment]
Bump. ^^

**Comment by SupaHam** [comment]
~@mibby I could add an extra argument to the message which would be the original name, not display name.~

Nevermind, I don't think that'd be a good idea.

**Comment by mibby** [comment]
@SupaHam Well in the meantime if toggling to a raw output isn't possible to have again, at least being able to recolor everything but the name is better than having it be the same format as normal messages.

**Comment by mibby** [comment]
I manually pulled https://github.com/drtshock/Essentials/pull/1460 for testing. While it isn't raw output and names are displayed based on the vault prefix/suffix, it is significantly better than it exists as now with it allowing the message text to be colored grey. Works for me. :)

Only issue I noticed is commands such as mail messages shown in socialspy are displayed white. Being able to re-color the socialspy mail msg command display to grey too would be nice as well.
